### ISSUE TITLE
Autosuggest key when creating TypedDict instances

### ISSUE THREAD

**Issue by mr-bjerre** [microsoft/pyright#1921]
I'm using VS Code with Pylance but I think the issue belongs here. Not sure though so forgive me if I'm wrong.

**Describe the bug**

1. When I am to create an instance that's typed as a TypedDict then I get no suggestions for the keys
1. I'm also getting a `Argument to class must be a base class (reportGeneralTypeIssues)` from pylance using `TypedDict` as base class (if I'm trying to import from `typing` first and falling back to importing from `typing_extensions`). If I only import from either `typing` or `typing_extensions` then I don't get the error

**To Reproduce**

Put the cursor inside `""` in the last line and press Shift + space for autosuggest

```python
try:
    from typing import TypedDict
except ImportError:
    from typing_extensions import TypedDict  # only available in typing from py3.8+


class Config(TypedDict):
    id: str


config: Config = {""}
```

**Expected behavior**

I'd expect VS Code to suggest adding `id` to the `dict`.

**VS Code extension or command-line**

I'm using pylance `v2021.5.4`

**Additional context**

The code completion works as expected when working with the `config` object and it also highlights when the creation of `config` is invalid - it just doesn't suggest the keys when "building" the `dict`

**Comment by erictraut** [comment]
Static type checkers cannot deal well with dynamic, conditional import constructs like the one you're using.

If your code needs to run on Python versions prior to Python 3.8, you should simply import from `typing_extensions` rather than doing a runtime try/except. That will eliminate the second error you're seeing in the type checker.

As for your feature request for completion suggestions within a dict literal statement, that's a duplicate of [this bug](https://github.com/microsoft/pylance-release/issues/654). Please upvote it because we prioritize new language server functionality based on user feedback.

**Comment by mr-bjerre** [comment]
Thanks for the quick and very good feedback! Enjoy your weekend
